src/commands/trigger.cpp

```diff
@@ -0,0 +1,104 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// trigger.cpp
+//
+// Identification: src/commands/trigger.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "commands/trigger.h"
+#include "parser/pg_trigger.h"
+
+namespace peloton {
+namespace commands {
+Trigger::Trigger(const peloton::planner::CreatePlan& plan) {
+  trigger_name = plan.GetTriggerName();
+  trigger_funcname = plan.GetTriggerFuncName();
+  trigger_args = plan.GetTriggerArgs();
+  trigger_columns = plan.GetTriggerColumns();
+  trigger_when = plan.GetTriggerWhen();
+  trigger_type = plan.GetTriggerType();
+}
+/*
+ * Add a trigger to the trigger list and update the summary
+ */
+void TriggerList::AddTrigger(Trigger trigger) {
+  triggers.push_back(trigger);
+  UpdateTypeSummary(trigger.GetTriggerType());
+}
+
+/*
+ * Update a type summary when a new type of trigger is added
+ */
+void TriggerList::UpdateTypeSummary(int16_t type) {
```

Comment:
Each line is in the same format like

`types_summary[XX] |= TRIGGER_TYPE_MATCHES(type, TRIGGER_TYPE_XX, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_XX);`

Since it contains much repeated work, can it be implemented using map? Let Map store parameters in .h file. (just personal suggestion, maybe my way makes code more confusing)